xarray/backends/api.py

```diff
@@ -562,12 +565,25 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
 
     if lock is None:
         lock = _default_lock(paths[0], engine)
-    datasets = [open_dataset(p, engine=engine, chunks=chunks or {}, lock=lock,
-                             **kwargs) for p in paths]
-    file_objs = [ds._file_obj for ds in datasets]
 
-    if preprocess is not None:
-        datasets = [preprocess(ds) for ds in datasets]
+    open_kwargs = dict(engine=engine, chunks=chunks or {}, lock=lock,
+                       autoclose=autoclose, **kwargs)
+
+    if parallel:
+        import dask.bag as db
```

Comment:
This is a minor point, but I find `dask.delayed()` slightly more general and easier to understand than `dask.bag`.

Reply:
I admit I don't fully understand the tradeoffs between these two approaches.

Reply:
They are basically equivalent. The only meaningful difference I think of aside from the API is that dask.delayed has fewer required dependencies (a subset of those required for dask.array):
https://github.com/dask/dask/blob/47d025476312c6a8155192196b8da3b5af0c13e4/setup.py#L10-L17